tools/base/fpgainfo/errors.c

```diff
@@ -0,0 +1,477 @@
+			uint64_t error_value = 0;
+
+			res = fpgaReadError(token, i, &error_value);
+			fpgainfo_print_err("reading error for FME", res);
+
+			printf("%-29s : 0x%" PRIX64 "\n", errinfos[i].name,
+			       error_value);
+
+			if (!strcmp(errinfos[i].name, "Errors")) {
+				size = FME_ERROR_COUNT;
+				error_string = FME_ERROR;
+			} else if (!strcmp(errinfos[i].name, "Next Error")) {
+				size = 0;
+				error_string = NULL;
+			} else if (!strcmp(errinfos[i].name, "First Error")) {
+				size = 0;
+				error_string = NULL;
+			} else if (!strcmp(errinfos[i].name, "PCIe0 Errors")) {
+				size = PCIE0_ERROR_COUNT;
+				error_string = PCIE0_ERROR;
+			} else if (!strcmp(errinfos[i].name, "Inject Error")) {
+				size = INJECT_ERROR_COUNT;
+				error_string = INJECT_ERROR;
+			} else if (!strcmp(errinfos[i].name,
+					   "Catfatal Errors")) {
+				size = CATFATAL_ERROR_COUNT;
+				error_string = CATFATAL_ERROR;
+			} else if (!strcmp(errinfos[i].name,
+					   "Nonfatal Errors")) {
+				size = NONFATAL_ERROR_COUNT;
+				error_string = NONFATAL_ERROR;
+			} else if (!strcmp(errinfos[i].name, "PCIe1 Errors")) {
+				size = PCIE1_ERROR_COUNT;
+				error_string = PCIE1_ERROR;
+			}
+
+			for (j = 0; (j < size) && (NULL != error_string); j++) {
+				if (FPGA_BIT_IS_SET(error_value, j)) {
+					printf("\t %s \n", error_string[j]);
+				}
+			}
+		}
+	} else if (((VERB_ALL == errors_config.which)
+		    || (VERB_PORT == errors_config.which))
+		   && (FPGA_ACCELERATOR == objtype)) {
+		fpgainfo_print_common("//****** PORT ERRORS ******//", props);
+
+		for (i = 0; i < (int)num_errors; i++) {
+			uint64_t error_value = 0;
+			res = fpgaReadError(token, i, &error_value);
+			fpgainfo_print_err("reading error for PORT", res);
+
+			printf("%-29s : 0x%" PRIX64 "\n", errinfos[i].name,
+			       error_value);
+
+			if (!strcmp(errinfos[i].name, "Errors")) {
+				size = PORT_ERROR_COUNT;
+				error_string = PORT_ERROR;
+			} else if (!strcmp(errinfos[i].name,
+					   "First Malformed Req")) {
+				size = 0;
+				error_string = NULL;
+			} else if (!strcmp(errinfos[i].name, "First Error")) {
+				size = 0;
+				error_string = NULL;
+			}
+
+			for (j = 0; (j < size) && (NULL != error_string); j++) {
+				if (FPGA_BIT_IS_SET(error_value, j)) {
+					printf("\t %s \n", error_string[j]);
+				}
+			}
+		}
+	}
+}
+
+fpga_result errors_command(fpga_token *tokens, int num_tokens, int argc,
+			   char *argv[])
+{
+	(void)argc;
+	(void)argv;
+	fpga_result res = FPGA_OK;
+	fpga_properties props;
+	struct fpga_error_info *errinfos = NULL;
+
+	if (errors_config.help_only) {
+		return res;
+	}
+
+	int i = 0;
+	for (i = 0; i < num_tokens; ++i) {
+		uint32_t num_errors;
+
+		res = fpgaGetProperties(tokens[i], &props);
+                if (res == FPGA_OK) {
+                        res = fpgaPropertiesGetNumErrors(props, &num_errors);
+                        fpgainfo_print_err("reading errors from properties", res);
+
+                        if (0 != num_errors) {
```

Comment:
No need to put 0 first. For comparisons using `==`, I think we've opted to do them the other way since modern  compilers will warn for potential assignment instead of comparison.